aten/src/ATen/native/mps/operations/View.mm

```diff
@@ -706,15 +685,27 @@ static IntArrayRef updateTensorBaseShape(const Tensor& self)
             }
             if (needsScatter) {
               auto updatesType = getMPSScalarType(updates.scalar_type());
-              newCachedGraph->updatesTensor = mpsGraphUnrankedPlaceHolder(mpsGraph, updatesType);
+              if ((updatesType == MPSDataTypeUInt8 || updatesType == MPSDataTypeBool) && !is_macos_13_or_newer()) {
+                  updatesType = MPSDataTypeInt8;
+              }
+              newCachedGraph->updatesTensor = mpsGraphRankedPlaceHolder(mpsGraph, updatesType, getMPSShape(self.numel()));
               updatesTensor = newCachedGraph->updatesTensor;
               if (inputType != updatesType) {
                 updatesTensor = [mpsGraph castTensor:updatesTensor
                                               toType:inputType
                                                 name:@"castUpdatesTensor"];
+                updatesType = inputType;
```

Comment:
This assignment doesn't seem to be used in the code that follows.

Reply:
Oh right, I used to pass it in chainViewOperation but I've removed that part since it was not needed.

Reply:
Updated